src/image/filter/__tests__/addSubtract.js

```diff
@@ -53,195 +54,170 @@ describe('subtract', function () {
 
 describe('subtract image', function () {
   it('should subtract an image to another', function () {
+
     let image = new Image(
       5,
       5,
-      [
-        0,
-        0,
-        255,
-        0,
-        255,
-        0,
-        0,
-        255,
-        0,
-        255,
-        0,
-        0,
-        255,
-        0,
-        255,
-        0,
-        0,
-        255,
-        0,
-        255,
-        0,
-        0,
-        255,
-        0,
-        255,
-      ],
+      array`
+        0, 0, 255, 0, 255
+        0, 0, 255, 0, 255
+        0, 0, 255, 0, 255
+        0, 0, 255, 0, 255
+        0, 0, 255, 0, 255
+      `,
       { kind: 'GREY' },
     );
 
     let image2 = new Image(
       5,
       5,
-      [
-        255,
-        0,
-        0,
-        0,
-        255,
-        255,
-        0,
-        0,
-        0,
-        255,
-        255,
-        0,
-        0,
-        0,
-        255,
-        255,
-        0,
-        0,
-        0,
-        255,
-        255,
-        0,
-        0,
-        0,
-        255,
-      ],
+      array`
+       255, 0, 0, 0, 255
+       255, 0, 0, 0, 255
+       255, 0, 0, 0, 255
+       255, 0, 0, 0, 255
+       255, 0, 0, 0, 255
+    `,
       { kind: 'GREY' },
     );
 
-    expect(Array.from(image.subtractImage(image2).data)).toStrictEqual([
-      0,
-      0,
-      255,
-      0,
-      0,
-      0,
-      0,
-      255,
-      0,
-      0,
-      0,
-      0,
-      255,
-      0,
-      0,
-      0,
-      0,
-      255,
-      0,
-      0,
-      0,
-      0,
-      255,
-      0,
-      0,
+    expect(Array.from(image.subtractImage(image2).data)).toStrictEqual([0, 0, 255, 0, 0, 0, 0, 255, 0, 0, 0, 0, 255, 0, 0, 0, 0, 255, 0, 0, 0, 0, 255, 0, 0,
     ]);
   });
+
+  it('should subtract a color image to another', function () {
+
+    let image = new Image(
+      1,
+      3,
+      array`
+        255, 0,   0,
+        0,   255, 0,
+        0,   0,   255,
+      `,
+      { kind: 'RGB' },
+    );
+
+    let image2 = new Image(
+      1,
+      3,
+      array`
+      127, 0,   0,
+      0,   127, 0,
+      0,   0,   127,
+    `,
+      { kind: 'RGB' },
+    );
+
+    expect(Array.from(image.subtractImage(image2).data)).toStrictEqual(
+      array`
+      128, 0,   0,
+      0,   128, 0,
+      0,   0,   128,
+    `
+    );
+  });
+
+  it('should subtract a color image to another with selected channels', function () {
```

Comment:
This test does not seem to be selecting channels. Actually it seems to be the exact same test as the previous one. Am I missing something?

Reply:
You are right this should be removed but moreover this part was already merged